arcade/types/rect.py

```diff
@@ -0,0 +1,511 @@
+        """
+        width = min(max_width or float("inf"), max(min_width or 0.0, self.width))
+        height = min(max_height or float("inf"), max(min_height or 0.0, self.height))
+        return self.resize(width, height, anchor)
+
+    def union(self, other: Rect) -> Rect:
+        """
+        Returns a new :py:class:`~arcade.types.rect.Rect` that is the union of this rect and another.
+        The union is the smallest rectangle that contains these two rectangles.
+        """
+        left = min(self.left, other.left)
+        right = max(self.right, other.right)
+        bottom = min(self.bottom, other.bottom)
+        top = max(self.top, other.top)
+        return LRBT(left, right, bottom, top)
+
+    def __or__(self, other: Rect) -> Rect:
+        return self.union(other)
+
+    def intersection(self, other: Rect) -> Rect | None:
+        """
+        Returns a new :py:class:`~arcade.types.rect.Rect` that is the overlaping portion of this Rect and another.
+        This will return None if no such rectangle exists.
+        """
+        intersecting = self.overlaps(other)
+        if not intersecting:
+            return None
+        left = max(self.left, other.left)
+        right = min(self.right, other.right)
+        bottom = max(self.bottom, other.bottom)
+        top = min(self.top, other.top)
+        return LRBT(left, right, bottom, top)
+
+    def __and__(self, other: Rect) -> Rect | None:
+        return self.intersection(other)
+
+    def overlaps(self, other: Rect) -> bool:
+        """
+        Returns True if `other` overlaps with the rect.
+        """
+
+        return (
+            (other.width + self.width) / 2.0 > abs(self.x - other.x)
+            and
+            (other.height + self.height) / 2.0 > abs(self.y - other.y)
+        )
+
+    def point_in_rect(self, point: Vec2) -> bool:
+        """Returns True if the given point is inside this rectangle."""
+        return (self.left < point.x < self.right) and (self.bottom < point.y < self.top)
+
+    def point_on_bounds(self, point: Vec2, tolerance: float) -> bool:
+        """Returns True if the given point is on the bounds of this rectangle within some tolerance."""
+        diff = Vec2(point.x - self.x, point.y - self.y)
+        dx = abs(diff.x) - self.width / 2.0
+        dy = abs(diff.y) - self.height / 2.0
+        d = (max(dx, 0.0)**2 + max(dy, 0.0)**2)**0.5 + min(max(dx, dy), 0.0)
+
+        return abs(d) < tolerance
+
+    def to_points(self) -> tuple[Vec2, Vec2, Vec2, Vec2]:
+        """Returns a tuple of the four corners of this Rect."""
+        left = self.left
+        bottom = self.bottom
+        right = self.right
+        top = self.top
+        return (
+            Vec2(left, bottom),
+            Vec2(left, top),
+            Vec2(right, top),
+            Vec2(right, bottom)
+        )
+
+    @property
+    def lbwh(self) -> RectParams:
+        """Provides a view into the Rect in the form of a tuple of (left, bottom, width, height)."""
```

Comment:
Provides a tuple in the format of (left, bottom, ...)

Reply:
I like this better, changed in latest commit.